boa/src/builtins/number/mod.rs

```diff
@@ -555,6 +555,188 @@ impl Number {
         }
     }
 
+    /// Builtin javascript 'isFinite(number)' function.
+    ///
+    /// Converts the argument to a number, throwing a type error if the conversion is invalid.
+    ///
+    /// If the number is NaN, +∞, or -∞ false is returned.
+    ///
+    /// Otherwise true is returned.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-isfinite-number
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isFinite
+    pub(crate) fn global_is_finite(
+        _this: &Value,
+        args: &[Value],
+        ctx: &mut Interpreter,
+    ) -> ResultValue {
+        if let Some(val) = args.get(0) {
+            let number = ctx.to_number(val)?;
+            Ok(Value::Boolean(number.is_finite()))
+        } else {
+            Ok(Value::Boolean(false))
```

Comment:
```suggestion
            Ok(number.is_finite().into())
        } else {
            Ok(false.into())
```
Prefer using the `from`/`into` functions over direct `enum` field access when constructing a `Value`, this makes it easier to refacor `Value` in the future, since we have a layer of abstraction.